mobile/src/utils/utils.ts

```diff
@@ -1,10 +1,47 @@
 import { AppState } from '../reducers/app.reducer';
 import { Employee } from '../reducers/organization/employee.model';
 import { Optional } from 'types';
+import { DayModel, IntervalModel, ReadOnlyIntervalsModel } from '../reducers/calendar/calendar.model';
+import { CalendarEvent } from '../reducers/calendar/calendar-event.model';
 
 //----------------------------------------------------------------------------
 export function getEmployee(state: AppState): Optional<Employee> {
     return (state.organization && state.userInfo && state.userInfo.employeeId) ?
         state.organization.employees.employeesById.get(state.userInfo.employeeId) :
         undefined;
 }
+
+//----------------------------------------------------------------------------
+function getVacationIntervals(intervals: IntervalModel[], exclude: CalendarEvent[] = []): IntervalModel[] {
+    return intervals
+        .filter(interval => interval.calendarEvent.isVacation)
```

Comment:
API will restrict all event types if they intersect with other events of the same type. I'm not sure it is correct, we need to ask master @santee.

Reply:
Hmmm. And what?

Reply:
What `what`? You application restricts only vacations to have intersected dates, API restricts sick leaves, workouts and day-offs also.

Reply:
"Restrict several vacations with intersected dates", and I believe that was you who created that issue. It also says "UI should not allow to select dates for new vacation that intersects with existing vacations (maybe, they should somehow be marked as disabled in calendar)." So, either you be more specific in issue description or be more specific here.

Reply:
Did I blame PR author that he can not read issue description?
I know that issue description is about vacations. I also know that I did server side PR with ALL calendar event types restricted. And moreover, my PR is not merged yet and I don't really know, should we do as you did, or as I did. That's why I need the answer of @santee to judge us.

Reply:
vacations must not intersect with each other, but they may intersect with other types - e.g. you can chose to work during vacation or create a sick leave while being on vacation.

Reply:
Yes, I meant that vacations should not intersect with vacations, sick leaves with sick leaves, and so on.

Reply:
22e5f4e63222c2157a937848ec3f4ea77e6367a9

Reply:
It's not possible to create intersecting dayoffs now. It's already not possible to create more than one sick leave.